lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1229,12 +1237,43 @@ def _on_logging_relation_changed(self, event: HookEvent):
                 charm must update its relation data.
         """
         if isinstance(event, RelationEvent):
+            self._set_endpoint_on_relation(event.relation)
             self._process_logging_relation_changed(event.relation)
         else:
             # Upgrade event or other charm-level event
             for relation in self._charm.model.relations[self._relation_name]:
+                self._set_endpoint_on_relation(relation)
                 self._process_logging_relation_changed(relation)
 
+    def _on_logging_relation_departed(self, event: RelationDepartedEvent):
+        """Removes alert rules files when consumer charms left the relation with Loki.
+
+        Args:
+            event: a `CharmEvent` in response to which the Loki
+                charm must update its relation data.
+        """
+        # self._set_endpoint_on_relation(event.relation)
```

Comment:
This commented-out line got me thinking: 
1. do we need a `if event.departing_unit == self.model.unit` guard if the charm is scaled down to zero units? I.e. is app data already handled correctly in that case?
1. do we need a `_set_endpoint_on_relation` or `_process_logging_relation_changed` on leader-elected?

Reply:
1.- As far as I understand it's not needed. App data is handled correctly while scaling down to zero and scaling up again to (let's say) 3 units.
2.- No, we do not need it since in the Provider side of the relation we need to set unit endpoint in every relation changed.